### PR TITLE
Add configuration service to JS SDK

### LINKED ISSUE #588: Add configuration service to JS SDK
#### Summary
#425 added an API for the cluster configuration. This should also be exposed as a service in the JS SDK. 

#### Why do we need this?
To access stack configuration data via the JS SDK. E.g. in #573, we need to retrieve the list of allowed frequency plans, to make the user choose from them.

#### What is already there? What do you see now?
#425

#### What is missing? What do you want to see?
A respective service in the SDK that connects to this API.

#### Environment
SDK in node or browser.

#### How do you propose to implement this?
Complementary to our other services.

#### Can you do this yourself and submit a Pull Request?
Yes.

### PR DESCRIPTION
**Pull request by kschiffer**
#### Summary
Closes #588
References: #573 

#### Changes
- Add a module with constants used throughout
- Add functionality to the api/http class to choose which component to query with an RPC based on the `config.component` value passed to the service function
  - The `Configuration.ListFrequencyPlans` rpc is exposed globally, but can return different values depending on what component it is queried against (NS gives frequency plans for device, GS for gateways)
- Add the configuration service

#### Release Notes
- Added configuration service to JS SDK

### CODE REVIEW

File: sdk/js/src/util/constants.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+const STACK_COMPONENTS = [ 'as', 'is', 'ns', 'js', 'gs' ]
+
+export {
+  STACK_COMPONENTS as default,
+  STACK_COMPONENTS,
+}
```
Comment (bafonins) [inline sdk/js/src/util/constants.js]: i guess this is done on purspose to avoid `eslint` warnings?

Comment (kschiffer) [reply sdk/js/src/util/constants.js]: Correct. Do you prefer an ignore comment?

Comment (bafonins) [reply sdk/js/src/util/constants.js]: no, its fine

Comment (kschiffer) [reply sdk/js/src/util/constants.js]: Fixed.

**Pull request merged**